### ISSUE TITLE
TypeError: Promise.prototype.then called on incompatible value  in Mozilla Firefox

### ISSUE THREAD

**Issue by BiggerCheng** [es-shims/Promise.prototype.finally#13]
use Promise.prototype.finally shim in Mozilla Firefox 56.0.2 (64 bit)  has  appear an error 

TypeError: Promise.prototype.then called on incompatible value

**Comment by ljharb** [comment]
Can you provide a bit more information? What code are you using the shim with, what code are you using to shim it?

**Comment by BiggerCheng** [comment]
![tim 20171109143919](https://user-images.githubusercontent.com/19924977/32592021-dc567c9a-c55c-11e7-97b0-5da8fac235d4.png)

**Comment by ljharb** [comment]
Thanks; I'm going to need more than a stack trace tho - can you include some of the original code, in `getInstanceFromGifFile`?

**Comment by BiggerCheng** [comment]
![image](https://user-images.githubusercontent.com/19924977/32592493-ec874dfe-c55e-11e7-8c8d-272609b2abc6.png)

**Comment by ljharb** [comment]
Thanks - and how did you install the `finally` shim? Do you have anything else that might be trying to polyfill `Promise`?

**Comment by BiggerCheng** [comment]
install this by webpack 3.6.0 ,babel 6.5.2,babel-loader 7.1.2 and l try polyfill Promise but the error also occur

**Comment by ljharb** [comment]
Sorry, let me rephrase. Where in your code do you include the finally shim, and what does that code look like?

**Comment by BiggerCheng** [comment]
![image](https://user-images.githubusercontent.com/19924977/32594759-30ad9dfe-c568-11e7-8738-08031e845605.png)

**Comment by BiggerCheng** [comment]
My Chrome  61.0.3163.100  does`t appear this error,it run fine ~

**Comment by ljharb** [comment]
Thanks. I'll see if I can reproduce it in Firefox.

**Comment by ljharb** [comment]
I'm unable to reproduce this in v56 or v57, with v3.1, v3.0.1, or v3.0.0 of the shim. Is it still an issue for you in Firefox 56 or 57?

**Comment by Akaryatrh** [comment]
Hi @ljharb, I'm experiencing the exact same issue with Firefox 61.0.1 (no issue in Chrome, didn't test Edge or IE11 yet)
```
TypeError: "Promise.prototype.then called on incompatible value"
finally_webpack-internal:///./node_modules/promise.prototype.finally/implementation.js:47:2
```
Here's how I'm importing the library:
```
import axios from 'axios'
import { shim } from 'promise.prototype.finally'
shim()
```

I'm using Quasar-cli that runs with the following dependencies:
```
"@babel/core": "7.0.0-beta.46",
"@babel/plugin-transform-runtime": "7.0.0-beta.46",
"@babel/preset-env": "7.0.0-beta.46",
"@babel/preset-stage-2": "7.0.0-beta.46",
"@babel/runtime": "7.0.0-beta.46",
"babel-loader": "8.0.0-beta.2",
"webpack": "4.8.3",
```
(full list is here: https://github.com/quasarframework/quasar-cli/blob/v0.16.0/package.json)

And I'm using:
```
"axios": "^0.18.0",
"promise.prototype.finally": "^3.1.0",
```

Any idea what could cause this error? Thx!

**Comment by ljharb** [comment]
If you're using babel-runtime, are you also using babel-polyfill, which includes core-js?

**Comment by Akaryatrh** [comment]
>are you also using babel-polyfill

Not that I'm aware of, and not presence either of this dependency in my `package.lock`.

**Comment by ljharb** [comment]
Yes, babel-polyfill is core-js.

core-js provides `finally`; while they should work together, it's not really necessary to have both.

**Comment by Akaryatrh** [comment]
I'm not sure to understand… You mean that it should work out of the box without the shim from `promise.prototype.finally`? If yes, it doesn't work (that's why I initially added it).

**Comment by ljharb** [comment]
I’d expect it to, yes - but if it doesn’t, then that makes sense why you’d have both :-)

I think you should be explicitly importing babel-polyfill somewhere, before you import the finally shim - rather than relying on preset-env to insert it.
